pilot/pkg/networking/core/v1alpha3/route/retry/retry.go

```diff
@@ -81,6 +81,10 @@ func ConvertPolicy(in *networking.HTTPRetry) *route.RetryPolicy {
 		// Allow the incoming configuration to specify both Envoy RetryOn and RetriableStatusCodes. Any integers are
 		// assumed to be status codes.
 		out.RetryOn, out.RetriableStatusCodes = parseRetryOn(in.RetryOn)
+		// If user has just specified HTTP status codes in retryOn but have not specified "retriable-status-codes", let us add it.
+		if len(out.RetriableStatusCodes) > 0 && !strings.Contains(out.RetryOn, "retriable-status-codes") {
+			out.RetryOn += ",retriable-status-codes"
```

Comment:
Isn't `RetryPolicy.RetriableStatusCodes` enough?

Reply:
It is not enough

See api docs 

https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_filters/router_filter.html?highlight=retriable%20status%20codes#x-envoy-retriable-status-codes

"Setting this header informs Envoy about what status codes should be considered retriable when used in conjunction with the retriable-status-code retry policy." - when used in conjunction with the retriable-status-code retry policy.

https://github.com/envoyproxy/envoy/blob/e071417f12809eddbf02052504fd9953eedce8f0/envoy/router/router.h#L245 (Envoy code)
https://github.com/envoyproxy/envoy/blob/main/source/common/router/retry_state_impl.cc#L345 - Considers retriable status code only when the policy is set

Reply:
good, reading the code is more clear.